pkg/pillar/cmd/zedagent/handleconfig.go

```diff
@@ -513,7 +516,63 @@ func inhaleDeviceConfig(config *zconfig.EdgeDevConfig, getconfigCtx *getconfigCo
 				ctx := getconfigCtx.zedagentCtx
 				triggerPublishDevInfo(ctx)
 			}
+		}
+		newEpoch := config.GetEpoch()
+		if epoch != newEpoch {
+			log.Functionf("Epoch changed from %d to %d", epoch, newEpoch)
+			epoch = newEpoch
+			ctx := getconfigCtx.zedagentCtx
+			// publish device info
+			triggerPublishDevInfo(ctx)
+
+			// publish applications infos
+			sub := ctx.getconfigCtx.subAppInstanceStatus
+			items := sub.GetAll()
+			for _, c := range items {
+				appStatus := c.(types.AppInstanceStatus)
+				uuidStr := appStatus.Key()
+				PublishAppInfoToZedCloud(ctx, uuidStr, &appStatus, ctx.assignableAdapters,
```

Comment:
This might take more than the 5 minute max causing a watchdog since each Send has a 2 minute time limit.
Thus you need to add a trigger logic for each object and have the main select loop pick up at most one trigger at a time.
Alternatively kick off a goroutine which will do all of the sends including the device info and exit when everything has been sent.